src/client/redux/user/user.actions.requests.js

```diff
@@ -48,29 +49,192 @@ export const listUsers = (page = 1, documentId, role) => async (dispatch) => {
+  const User = new UserService();
+  setIsLoading(dispatch, true);
+  const {
+    firstName,
+    lastName,
+    birthDate,
+    phoneNumber,
+    country,
+    gender,
+    role,
+    email,
+    documentId,
+  } = data;
+  const newUser = {
+    profile: {
+      firstName,
+      lastName,
+      birthDate,
+      phoneNumber,
+      avatar: '',
+      avatarMimeType: '',
+      gender,
+      country,
+      documentId,
+    },
+    auth: {
+      email,
+      role,
+    },
+  };
+  try {
+    await User.newUser(newUser);
+    await dispatch(listUsers());
     dispatch(feedbackActions.setFeedback({
       feedback: {
-        message: error.message,
-        type: 'error',
+        message: 'Usuario creado exitosamente',
+        type: 'success',
       },
     }));
+  } catch (error) {
+    dispatch(feedbackActions.setFeedback({
+      message: error.message,
+      type: 'error',
+    }));
+  } finally {
     setIsLoading(dispatch, false);
-    throw error;
   }
-  setIsLoading(dispatch, false);
+};
+
+export const updateUser = (data) => async (dispatch) => {
+  const User = new UserService();
+  setIsLoading(dispatch, true);
+  const {
+    firstName,
+    lastName,
+    birthDate,
+    phoneNumber,
+    country,
+    gender,
+    email,
+    id,
+  } = data;
+  const updatedUser = {
+    profile: {
+      firstName,
+      lastName,
+      birthDate,
+      phoneNumber,
+      avatar: '',
+      gender,
+      country,
+    },
+    auth: {
+      email,
+    },
+  };
+  try {
+    await User.updateUser(updatedUser, id);
+    await dispatch(listUsers());
+    dispatch(feedbackActions.setFeedback({
+      feedback: {
+        message: 'Usuario actualizado exitosamente',
+        type: 'success',
+      },
+    }));
+  } catch (error) {
+    dispatch(feedbackActions.setFeedback({
+      message: error.message,
+      type: 'error',
+    }));
+  } finally {
+    setIsLoading(dispatch, false);
+  }
+};
+
+export const deleUser = (id) => async (dispatch) => {
+  const User = new UserService();
+  setIsLoading(dispatch, true);
+  try {
+    await User.deleteUser(id);
+    await dispatch(listUsers());
+    dispatch(feedbackActions.setFeedback({
+      feedback: {
+        message: 'Usuario eliminado correctamente',
+        type: 'success',
+      },
+    }));
+  } catch (error) {
+    dispatch(feedbackActions.setFeedback({
+      message: error.message,
+      type: 'error',
+    }));
+  } finally {
+    setIsLoading(dispatch, false);
+  }
+};
+
+export const updateProfile = (data) => async (dispatch) => {
+  const User = new UserService();
+  setIsLoading(dispatch, true);
+  const {
+    firstName,
+    lastName,
+    birthDate,
+    phone,
+    avatar,
+    country,
+    id,
+    gender,
+    email,
+  } = data;
+
+  const userProfile = {
+    profile: {
+      firstName,
+      lastName,
+      birthDate,
+      phoneNumber: phone,
+      avatar,
+      gender,
+      country,
+      email,
+    },
+  };
```

Comment:
From line 195 to 218 should be in the user service, the reducer should not have that responsibility.

Reply:
Modified, there were two other similar cases and they were changed.